src/imars3d/backend/workflow/engine.py

```diff
@@ -32,27 +35,22 @@ def _validate_outputs(function_outputs: Union[None, Iterable[Any]]):
         ------
         WorkflowEngineError
         """
-        if function_outputs is None:
-            return
-        tests = {
-            "Return value must be an iterable ": hasattr(function_outputs, "__iter__"),
-            "Return value cannot be a string": not isinstance(function_outputs, str),
-        }
-        errors = " and ".join([k for k, v in tests.items() if v is False])
-        if errors:
-            raise WorkflowEngineError(errors)
-
-    def __init__(self, global_params: Union[Set[str], List[str]]) -> None:
-        r"""
 
-        Parameters
-        ----------
-        global_params
-            List of JSON entries deemed as accessible by any workflow task. These are the metadata
-            and the task outputs.
-        """
-        self._global_params: set = set(global_params)
-        self._registry: Optional[dict] = None  # will store set or computed globals parameters
+        def validate_type(outputs):
+            if not hasattr(outputs, "__iter__"):
```

Comment:
[Here's the post](https://stackoverflow.com/questions/1952464/in-python-how-do-i-determine-if-an-object-is-iterable) that led me to use `__iter__` instead of `Iterable`, in particular at the end it says:

> The collections module provides some abstract base classes, which allow to ask classes or instances if they provide particular functionality, for example:
> 
>  from collections.abc import Iterable
> 
>  if isinstance(e, Iterable):
>      # e is iterable
> However, this does not check for classes that are iterable through __getitem__.